### PR TITLE
WAC channel

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/channels/types/whatsapp_cloud/tests.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+            response = self.client.post(connect_whatsapp_cloud_url, dict(user_access_token="X" * 36), follow=True)
+            self.assertEqual(response.status_code, 200)
+
+        # make sure the token is set on the session
+        session = self.client.session
+        session[Channel.CONFIG_WHATSAPP_CLOUD_USER_TOKEN] = "user-token"
+        session.save()
+
+        self.assertTrue(Channel.CONFIG_WHATSAPP_CLOUD_USER_TOKEN in self.client.session)
```
Comment (rowanseymour) [inline temba/channels/types/whatsapp_cloud/tests.py:127]: `self.assertIn`

File: temba/msgs/models.py [hunk 2]
```diff
@@ -892,6 +892,8 @@ class Label(LegacyUUIDMixin, TembaModel, DependencyMixin):
 
     org_limit_key = Org.LIMIT_LABELS
 
+    org_limit_key = Org.LIMIT_LABELS
```
Comment (rowanseymour) [inline temba/msgs/models.py:895]: accidental keyboard hit?

File: temba/request_logs/models.py [hunk 3]
```diff
@@ -150,6 +153,33 @@ def create_from_exception(cls, log_type, url, exception, start, classifier=None,
...
+
+    def get_response_display(self):
+        """
+        Gets the response trace as it should be displayed to users
+        """
+        return self._get_display_value(self.response, ContactURN.ANON_MASK)
+
+    def _get_display_value(self, original, mask):
+
+        secrets = [settings.WHATSAPP_ADMIN_SYSTEM_USER_TOKEN]
```
Comment (rowanseymour) [inline temba/request_logs/models.py:176]: This feels like it's in the wrong place. We use this class to represent request logs for all sorts of things, 99% of which won't contain a whatsapp token. So it feels inefficient that we'll look for that in every single request URL, payload and response.

Maybe pass in a list of things to redact as an arg?

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: temba/request_logs/models.py (hunk 3 above)
Comment (norkans7) [reply temba/request_logs/models.py:176]: This method is called in the 3 methods above, I guess we can use a if block here instead to make sure we have the correct list to redact when the log type is one of 
`WHATSAPP_TEMPLATES_SYNCED, WHATSAPP_TOKENS_SYNCED, WHATSAPP_CONTACTS_REFRESHED, WHATSAPP_CHECK_HEALTH`

Comment (norkans7) [reply temba/request_logs/models.py:176]: This method is called in the 3 methods above, I guess we can use a if block here instead to make sure we have the correct list to redact when the log type is one of 
`WHATSAPP_TEMPLATES_SYNCED, WHATSAPP_TOKENS_SYNCED, WHATSAPP_CONTACTS_REFRESHED, WHATSAPP_CHECK_HEALTH`

File: temba/request_logs/models.py [hunk 4]
```diff
@@ -150,6 +153,43 @@ def create_from_exception(cls, log_type, url, exception, start, classifier=None,
...
+        Gets the response trace as it should be displayed to users
+        """
+        return self._get_display_value(self.response, ContactURN.ANON_MASK)
+
+    def _get_display_value(self, original, mask):
+
+        redact_secrets = self.get_redact_secrets()
+
+        for secret in redact_secrets:
+            if secret and original:
```
Comment (rowanseymour) [inline temba/request_logs/models.py:189]: this `if` feels a little unnecessary

File: templates/channels/types/whatsapp_cloud/claim.haml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  .mt-4.card#fb-channel-options
+    %p(style="font-size:1rem;")
+      -trans "Select the number you want to add as a channel:"
+    
+    -for phone_num in phone_numbers
+      .fb-page-channel-option.lbl.mt-3.mr-2.p-2.linked(class="" data-number="{{phone_num.display_phone_number}}" data-verified_name="{{phone_num.verified_name}}" data-phone_number_id="{{phone_num.phone_number_id}}" data-waba_id="{{phone_num.waba_id}}" data-business_id="{{phone_num.business_id}}" data-currency="{{phone_num.currency}}" data-message_template_namespace="{{phone_num.message_template_namespace}}")
+        {{phone_num.display_phone_number}} ({{phone_num.verified_name}})
+
+
+
```
Comment (rowanseymour) [inline templates/channels/types/whatsapp_cloud/claim.haml:35]: less spaces...

File: temba/channels/models.py [hunk 6]
```diff
@@ -1244,6 +1249,10 @@ def _get_display_value(self, user, original, mask, redact_keys=()):
         """
         Get a part of the log which may or may not have to be redacted to hide sensitive information in anon orgs
         """
+        secrets = [settings.WHATSAPP_ADMIN_SYSTEM_USER_TOKEN]
+        for secret in secrets:
+            if secret and original:
+                original = redact.text(original, secret, mask)
```
Comment (rowanseymour) [inline temba/channels/models.py:1255]: this again feels pretty weird - this method is used across all channel log types and takes a set of things to redact so why wouldn't we pass `settings.WHATSAPP_ADMIN_SYSTEM_USER_TOKEN` in with those?

also the `if` isn't necessary

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: temba/channels/models.py (hunk 6 above)
Comment (norkans7) [reply temba/channels/models.py:1256]: Added the if block above to only use this for WAC channel logs

Comment (rowanseymour) [reply temba/channels/models.py:1256]: This still doesn't feel right - the whole reason for channel types is that they're pluggable and `temba.channels` doesn't need to know about specific types.

ChannelType already has a `redact_request_keys` and `redact_response_keys`.. maybe it needs `redact_values` as well?

Comment (norkans7) [reply temba/channels/models.py]: Added that now in https://github.com/nyaruka/rapidpro/pull/3819/commits/cd64667fa0623f32dc04f508209af848c3b7e5ed

**Pull request merged**